enterprise/admin/views.py

```diff
@@ -291,132 +292,396 @@ def _handle_bulk_upload(cls, enterprise_customer, manage_learners_form, request,
-                "The following users do not have an account on {}. They have not been enrolled in the course."
-                " When these users create an account, they will be enrolled in the course automatically: {}"
-            ).format(settings.PLATFORM_NAME, ", ".join(non_existing)))
-            for email in non_existing:
-                pending_user = PendingEnterpriseCustomerUser.objects.get(
+    @classmethod
+    def program_notify(cls, enterprise_customer, program_details, users):
+        """
+        Notify learners about a program in which they've been enrolled.
+
+        Args:
+            enterprise_customer: The EnterpriseCustomer being linked to
+            program_details: Details about the specific program the learners were enrolled in
+            users: An iterable of the users or pending users who were enrolled
+        """
+        program_name = program_details.get('title')
+        program_branding = program_details.get('type')
+        program_url = program_details.get('marketing_url')
+        program_type = 'program'
+        program_start = get_earliest_start_date_from_program(program_details)
+
+        with mail.get_connection() as email_conn:
+            for user in users:
+                send_email_notification_message(
+                    user=user,
+                    enrolled_in={
+                        'name': program_name,
+                        'url': program_url,
+                        'type': program_type,
+                        'start': program_start,
+                        'branding': program_branding,
+                    },
                     enterprise_customer=enterprise_customer,
-                    user_email=email
+                    email_connection=email_conn
                 )
-                PendingEnrollment.objects.update_or_create(
-                    user=pending_user,
+
+    @classmethod
+    def get_success_message(cls, users, enrolled_in):
+        """
+        Create messages for the users who were enrolled in a course or program.
+
+        Args:
+            users: An iterable of users who were successfully enrolled
+            enrolled_in (str): A string identifier for the course or program the users were enrolled in
+
+        Returns:
+            tuple: A 2-tuple containing a message type and message text
+        """
+        enrolled_count = len(users)
+        return (
+            'success',
+            ungettext(
+                '{enrolled_count} user was enrolled to {enrolled_in}.',
+                '{enrolled_count} users were enrolled to {enrolled_in}.',
+                enrolled_count,
+            ).format(
+                enrolled_count=enrolled_count,
+                enrolled_in=enrolled_in,
+            )
+        )
+
+    @classmethod
+    def get_failed_message(cls, users, enrolled_in):
+        """
+        Create messages for the users who were not able to be enrolled in a course or program.
```

Comment:
Again, we're just creating a single "failed" message given a set of users and a course/program id -- this function doesn't really have context into how/why it's being used.